Backend/utils/cronJobTrendingArtists.js

```diff
@@ -31,66 +32,113 @@ async function calculateTrendingArtists() {
+        artistScores[artistName].tagCount += 1;
 
         const artist = await prisma.artist.findFirst({
             where: { name: artistName }
         });
 
         if (artist) {
-            artistMomentum[artistName].popularity = artist.popularity;
-            artistMomentum[artistName].artistId = artist.id;
+            artistScores[artistName].popularity = artist.popularity;
+            artistScores[artistName].artistId = artist.id;
         }
     }
 
-    // Add search count momentum
+    // Add search count score
     for (const search of recentSearches) {
         const artistName = search.artist.name;
-        if (!artistMomentum[artistName]) {
-            artistMomentum[artistName] = {
+        if (!artistScores[artistName]) {
+            artistScores[artistName] = {
                 tagCount: 0,
                 popularity: 0,
                 searchCount: 0,
                 artistId: search.artist.id,
             };
         }
-        artistMomentum[artistName].searchCount += 1;
+        artistScores[artistName].searchCount += 1;
     }
 
-    // Combine tag count, search count, and popularity to calculate momentum
-    const dailyTrendingArtists = Object.entries(artistMomentum)
+    const dailyTrendingArtists = Object.entries(artistScores)
         .map(([name, data]) => ({
             artistId: data.artistId,
-            momentum: data.tagCount + data.searchCount + data.popularity * popularityWeight
-        }))
-        .sort((a, b) => b.momentum - a.momentum);
+            popularityScore: data.tagCount + data.searchCount + data.popularity * popularityWeight
+        }));
 
-    // Store daily trending artists in the database with the current date
+    // Store daily popularity scores in the database with the current date
     for (const artist of dailyTrendingArtists) {
         if (artist.artistId) {
             await prisma.trendingArtist.create({
                 data: {
                     artistId: artist.artistId,
-                    momentum: artist.momentum,
-                    date: oneDayAgo // store the date for historical tracking
+                    popularityScore: artist.popularityScore,
+                    date: oneDayAgo, // store the date for historical tracking
+                    momentum: 0 // Initialize momentum to 0 for now
                 }
             });
         }
     }
+
+    // Fetch popularity scores from the last two weeks
+    const trendingArtistsTwoWeeks = await prisma.trendingArtist.findMany({
+        where: {
+            date: {
+                gte: twoWeeksAgo
+            }
+        },
+        orderBy: {
+            date: 'asc'
+        }
+    });
+
+    // Calculate momentum by comparing scores
+    const momentumData = {};
+
+    for (const record of trendingArtistsTwoWeeks) {
+        if (!momentumData[record.artistId]) {
+            momentumData[record.artistId] = [];
+        }
+        momentumData[record.artistId].push(record.popularityScore);
```

Comment:
just making sure I understand correctly - when you say "renamed momentum to popularity score" you are actually using both terms now, but popularity is calculated on a daily basis whereas momentum is over the course of the last 2 weeks?

Reply:
That's correct!